acareeg/eegip.py

```diff
@@ -171,53 +175,122 @@ def process_events_london_resting_state(raw, age, tmax=1):
 
         if len(rst0) == 0 and len(rst1) == 0:
             return None
+        assert(len(rst0) == len(rst1))
 
-        annot_sample = np.concatenate([np.arange(start, stop - 0.999, 1.0) for start, stop in zip(rst0, rst1)])
-        annot_sample = (annot_sample * freq).astype(int)
+        # Keep only trials that are at least 10 s long.
+        # All trials are defined by Rst0-Rst1 pair of events.
+        annot_sample = np.concatenate([np.arange(start*freq, (stop-tmax)*freq, tmax*freq)
+                                       for start, stop in zip(rst0, rst1)
+                                       if stop - start > 10.0]).astype(int)
         annot_id = [1] * len(annot_sample)
```

Comment:
I dont understand what information len(annot_sample) is giving us

Reply:
Annot sample will be a list of the sample number that start every event. For a start at sample 0, a end at sample 2000, a tmax=1s, and a freq=500 Hz, that would be the list [0, 500, 1000, 1500]. The length of this list gives you the numbers of events that will be returned... so annot_id will be a list [1 1 1 1 1 ...] of the same length as the number of events.